pkg/compare/compare.go

```diff
@@ -90,11 +90,56 @@ func YAMLCmpWithIgnore(a, b string, ignorePaths []string) string {
 	if err := yaml.Unmarshal([]byte(b), &bo); err != nil {
 		return err.Error()
 	}
+
+	unmarshalConfigMapData(ao)
+	unmarshalConfigMapData(bo)
+
 	var r YAMLCmpReporter
 	cmp.Equal(ao, bo, cmp.Reporter(&r), genPathIgnoreOpt(ignorePaths))
 	return r.String()
 }
 
+func unmarshalConfigMapData(obj map[string]interface{}) {
+	InlineYamlData(obj, "ConfigMap", "data")
+}
+
+func InlineYamlData(obj map[string]interface{}, targetKind, targetPath string) {
+	if kind, ok := obj["kind"]; !ok || kind != targetKind {
+		return
+	}
+
+	nodeList := strings.Split(targetPath, ".")
+	if len(nodeList) == 0 {
+		return
+	}
+
+	cur := obj
+	for _, nname := range nodeList {
+		ndata, ok := cur[nname]
+		if !ok || ndata == nil {
+			return // target path does not exist
```

Comment:
maybe helpful to add logging messages for some return cases?

Reply:
will return an error and add logs.